src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexFindOptimizations.cs

```diff
@@ -571,34 +593,35 @@ public bool TryFindNextStartingPosition(ReadOnlySpan<char> textSpan, ref int pos
 
                 case FindNextStartingPositionMode.FixedDistanceSets_LeftToRight:
                     {
-                        List<(char[]? Chars, string Set, int Distance)> sets = FixedDistanceSets!;
-                        (char[]? primaryChars, string primarySet, int primaryDistance) = sets[0];
+                        List<FixedDistanceSet> sets = FixedDistanceSets!;
+                        FixedDistanceSet primarySet = sets[0];
+
                         int endMinusRequiredLength = textSpan.Length - Math.Max(1, MinRequiredLength);
 
-                        if (primaryChars is not null)
+                        if (primarySet.Chars is not null)
```

Comment:
NIT: would it be a tiny bit better to cache the member access into a local and use that throughout the method?

Reply:
From a perf perspective, it shouldn't matter (Chars is a public field on the struct).  I'll leave it as-is for now. Thanks.